fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java

```diff
@@ -355,7 +365,34 @@ public void afterSearch(FHIRPersistenceEvent event) throws FHIRPersistenceInterc
     }
 
     /**
-     * Enforce the authorizations granted by the end user in the form of scope strings
+     * Check whether the permissions required for the current interaction are granted by the approved scopes.
+     *
+     * @param resourceType the resource type used in the endpoint of the request (or "Resource" for whole-system interactions)
+     * @param requiredPermission the permission required for the requested interaction
+     * @param approvedScopes a list of SMART scopes associated with the request
+     * @throws FHIRPersistenceInterceptorException if the interaction is not permitted
+     */
+    private void checkScopes(String resourceType, Permission requiredPermission, List<Scope> approvedScopes) throws FHIRPersistenceInterceptorException {
```

Comment:
Seems like this does a subset of what the isAllowed() method is doing - wondering if there's a way to factor out the common code into one method?

Reply:
Yeah, I initially started there but in this case we just have to loop until we confirm the interaction is allowed whereas in the other one we organize the scopes for later use.